### ISSUE TITLE
Suggestion: Rethink Device and Device State

### ISSUE THREAD

**Issue by Abyss777** [tananaev/traccar#3742]
We have dynamic Device State objects that trace motion and overspeeding and mixed (static and dynamic) Device objects. Devices partially saved in DB and partially used to trace Connection Status and current geofences(`geofenceIds`). I think it is logical to move dynamic part to Device State.

I would suggest next:
- Move `status` to DeviceState model
- Move `geofenceIds` to DeviceState model
- Send only DeviceState instead of whole Device via websocket
- Add API to get current DeviceState if necessary

I see next benefits:
- Device model is not overloaded by dynamic fields
- Less amount of data sent via websocket (device model might be rather large with descriptions, attributes and others)
- Allow web application see motion and overspeed state from backend. (There are dozens requests how to show motion state on device icon instead of connection status)
- Future base place for tracing other dynamic states, like harsh driving or something.

And I see next drawbacks:
- We need combine Device and DeviceState on web side to show in Devices panel. It is not problem, just more complex.

What do you think?

**Comment by tananaev** [comment]
Makes sense to me. I also agree with drawback, so if you can find a way to implement it in a nice simple way please go ahead.

**Comment by Abyss777** [comment]
Hm... looks like not much saved data in websocket connection.
We are updating `lastUpdate` Device field on every incoming message and disconnection, therefore we still need resend whole Device model via websocket.

I see two options, leave everything as is or internally move `lastUpdate` to `DeviceState`, but update in DB as `lastUpdate` column of `devices` table.
Any ideas are appreciated...

**Comment by tananaev** [comment]
One option I can think of is to basically duplicate `lastUpdate` in both device and device state.
